route/build/src/historyFile.f90

```diff
@@ -30,14 +27,14 @@ MODULE historyFile
 
   type :: histFile
     private
-    character(FileStrLen) :: fname               ! netCDF name
-    integer(i4b)          :: iTime=0             ! time step in output netCDF
-    logical(lgt)          :: fileStatus=.false.  ! flag to indicate history output netcdf is open
-    logical(lgt)          :: gageOutput=.false.  ! flag to indicate this is at-gage-only output (== output subset of reaches)
-    type(iosystem_desc_t) :: pioSys              ! PIO system (this does not have to be initialize for each history file?)
-    type(file_desc_t)     :: pioFileDesc         ! PIO data identifying the file
-    type(io_desc_t)       :: ioDescRchFlux       ! PIO domain decomposition data for reach flux [nRch]
-    type(io_desc_t)       :: ioDescHruFlux       ! PIO domain decomposition data for hru runoff [nHRU]
+    character(FileStrLen)          :: fname               ! netCDF name
+    integer(i4b)                   :: iTime=0             ! time step in output netCDF
+    logical(lgt)                   :: fileStatus=.false.  ! flag to indicate history output netcdf is open
+    logical(lgt)                   :: gageOutput=.false.  ! flag to indicate this is at-gage-only output (== output subset of reaches)
+    type(iosystem_desc_t),pointer  :: pioSys              ! PIO system (this does not have to be initialize for each history file?)
```

Comment:
I suggest removing pioSys from here, but use pioSystem from globaldata, and then make pioSystem in globalData a pointer, and change RtmMod so that you set pioSystem as a pointer like this...

    pioSystem         => shr_pio_getiosys(inst_name)

Reply:
pioSys has been removed from this type definition and use pioSystem directly from globalData